# Importing Libraries

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')


# Reading Dataset

In [2]:
df=pd.read_csv(r'C:\Users\Nikhil\Documents\world_population.csv')

In [3]:
df

Rank CCA3  Country/Territory           Capital Continent  \
0      36  AFG        Afghanistan             Kabul      Asia   
1     138  ALB            Albania            Tirana    Europe   
2      34  DZA            Algeria           Algiers    Africa   
3     213  ASM     American Samoa         Pago Pago   Oceania   
4     203  AND            Andorra  Andorra la Vella    Europe   
..    ...  ...                ...               ...       ...   
229   226  WLF  Wallis and Futuna          Mata-Utu   Oceania   
230   172  ESH     Western Sahara          El Aaiún    Africa   
231    46  YEM              Yemen             Sanaa      Asia   
232    63  ZMB             Zambia            Lusaka    Africa   
233    74  ZWE           Zimbabwe            Harare    Africa   

     2022 Population  2020 Population  2015 Population  2010 Population  \
0           41128771         38972230         33753499         28189672   
1            2842321          2866849          2882481          2913399   
2           44903225         43451666         39543154         35856344   
3              44273            46189            51368            54849   
4              79824            77700            71746            71519   
..               ...              ...              ...              ...   
229            11572            11655            12182            13142   
230           575986           556048           491824           413296   
231         33696614         32284046         28516545         24743946   
232         20017675         18927715         16248230         13792086   
233         16320537         15669666         14154937         12839771   

     2000 Population  1990 Population  1980 Population  1970 Population  \
0           19542982         10694796         12486631         10752971   
1            3182021          3295066          2941651          2324731   
2           30774621         25518074         18739378         13795915   
3              58230            47818            32886            27075   
4              66097            53569            35611            19860   
..               ...              ...              ...              ...   
229            14723            13454            11315             9377   
230           270375           178529           116775            76371   
231         18628700         13375121          9204938          6843607   
232          9891136          7686401          5720438          4281671   
233         11834676         10113893          7049926          5202918   

     Area (km²)  Density (per km²)  Growth Rate  World Population Percentage  
0        652230            63.0587       1.0257                         0.52  
1         28748            98.8702       0.9957                         0.04  
2       2381741            18.8531       1.0164                         0.56  
3           199           222.4774       0.9831                         0.00  
4           468           170.5641       1.0100                         0.00  
..          ...                ...          ...                          ...  
229         142            81.4930       0.9953                         0.00  
230      266000             2.1654       1.0184                         0.01  
231      527968            63.8232       1.0217                         0.42  
232      752612            26.5976       1.0280                         0.25  
233      390757            41.7665       1.0204                         0.20  

[234 rows x 17 columns]

In [4]:
df.columns

Index(['Rank', 'CCA3', 'Country/Territory', 'Capital', 'Continent',
       '2022 Population', '2020 Population', '2015 Population',
       '2010 Population', '2000 Population', '1990 Population',
       '1980 Population', '1970 Population', 'Area (km²)', 'Density (per km²)',
       'Growth Rate', 'World Population Percentage'],
      dtype='object')

# Droping columns

In [5]:
df.drop(['CCA3','Country/Territory','Capital','1990 Population','1980 Population', '1970 Population', 'Area (km²)', 'Density (per km²)',],axis=1,inplace=True)

In [6]:
df

Rank Continent  2022 Population  2020 Population  2015 Population  \
0      36      Asia         41128771         38972230         33753499   
1     138    Europe          2842321          2866849          2882481   
2      34    Africa         44903225         43451666         39543154   
3     213   Oceania            44273            46189            51368   
4     203    Europe            79824            77700            71746   
..    ...       ...              ...              ...              ...   
229   226   Oceania            11572            11655            12182   
230   172    Africa           575986           556048           491824   
231    46      Asia         33696614         32284046         28516545   
232    63    Africa         20017675         18927715         16248230   
233    74    Africa         16320537         15669666         14154937   

     2010 Population  2000 Population  Growth Rate  \
0           28189672         19542982       1.0257   
1            2913399          3182021       0.9957   
2           35856344         30774621       1.0164   
3              54849            58230       0.9831   
4              71519            66097       1.0100   
..               ...              ...          ...   
229            13142            14723       0.9953   
230           413296           270375       1.0184   
231         24743946         18628700       1.0217   
232         13792086          9891136       1.0280   
233         12839771         11834676       1.0204   

     World Population Percentage  
0                           0.52  
1                           0.04  
2                           0.56  
3                           0.00  
4                           0.00  
..                           ...  
229                         0.00  
230                         0.01  
231                         0.42  
232                         0.25  
233                         0.20  

[234 rows x 9 columns]

# Checking null value

In [7]:
df.isna().sum()

Rank                           0
Continent                      0
2022 Population                0
2020 Population                0
2015 Population                0
2010 Population                0
2000 Population                0
Growth Rate                    0
World Population Percentage    0
dtype: int64

# Checking unique Values

In [8]:
df['Continent'].unique()

array(['Asia', 'Europe', 'Africa', 'Oceania', 'North America',
       'South America'], dtype=object)

# Replacing value

In [9]:
df.Continent=df.Continent.replace({'Asia':1,'Europe':2,'Africa':3,'Oceania':4,'North America':5,'South America':6})


In [10]:
df

Rank  Continent  2022 Population  2020 Population  2015 Population  \
0      36          1         41128771         38972230         33753499   
1     138          2          2842321          2866849          2882481   
2      34          3         44903225         43451666         39543154   
3     213          4            44273            46189            51368   
4     203          2            79824            77700            71746   
..    ...        ...              ...              ...              ...   
229   226          4            11572            11655            12182   
230   172          3           575986           556048           491824   
231    46          1         33696614         32284046         28516545   
232    63          3         20017675         18927715         16248230   
233    74          3         16320537         15669666         14154937   

     2010 Population  2000 Population  Growth Rate  \
0           28189672         19542982       1.0257   
1            2913399          3182021       0.9957   
2           35856344         30774621       1.0164   
3              54849            58230       0.9831   
4              71519            66097       1.0100   
..               ...              ...          ...   
229            13142            14723       0.9953   
230           413296           270375       1.0184   
231         24743946         18628700       1.0217   
232         13792086          9891136       1.0280   
233         12839771         11834676       1.0204   

     World Population Percentage  
0                           0.52  
1                           0.04  
2                           0.56  
3                           0.00  
4                           0.00  
..                           ...  
229                         0.00  
230                         0.01  
231                         0.42  
232                         0.25  
233                         0.20  

[234 rows x 9 columns]

# Checking datatypes

In [11]:
df.dtypes

Rank                             int64
Continent                        int64
2022 Population                  int64
2020 Population                  int64
2015 Population                  int64
2010 Population                  int64
2000 Population                  int64
Growth Rate                    float64
World Population Percentage    float64
dtype: object

# Converting datatypes to integer

In [12]:
df['Growth Rate']=df['Growth Rate'].astype(int)
df['World Population Percentage']=df['World Population Percentage'].astype(int)

In [13]:
df.dtypes

Rank                           int64
Continent                      int64
2022 Population                int64
2020 Population                int64
2015 Population                int64
2010 Population                int64
2000 Population                int64
Growth Rate                    int32
World Population Percentage    int32
dtype: object

# Stastical information

In [14]:
df.describe()

Rank   Continent  2022 Population  2020 Population  \
count  234.000000  234.000000     2.340000e+02     2.340000e+02   
mean   117.500000    2.978632     3.407441e+07     3.350107e+07   
std     67.694165    1.548768     1.367664e+08     1.355899e+08   
min      1.000000    1.000000     5.100000e+02     5.200000e+02   
25%     59.250000    2.000000     4.197385e+05     4.152845e+05   
50%    117.500000    3.000000     5.559944e+06     5.493074e+06   
75%    175.750000    4.000000     2.247650e+07     2.144798e+07   
max    234.000000    6.000000     1.425887e+09     1.424930e+09   

       2015 Population  2010 Population  2000 Population  Growth Rate  \
count     2.340000e+02     2.340000e+02     2.340000e+02   234.000000   
mean      3.172996e+07     2.984524e+07     2.626947e+07     0.811966   
std       1.304050e+08     1.242185e+08     1.116982e+08     0.391577   
min       5.640000e+02     5.960000e+02     6.510000e+02     0.000000   
25%       4.046760e+05     3.931490e+05     3.272420e+05     1.000000   
50%       5.307400e+06     4.942770e+06     4.292907e+06     1.000000   
75%       1.973085e+07     1.915957e+07     1.576230e+07     1.000000   
max       1.393715e+09     1.348191e+09     1.264099e+09     1.000000   

       World Population Percentage  
count                   234.000000  
mean                      0.256410  
std                       1.624141  
min                       0.000000  
25%                       0.000000  
50%                       0.000000  
75%                       0.000000  
max                      17.000000

# Seprating Dependent and Indeoendent Variable

In [15]:
x=df.drop(['Continent'],axis=1)
x


Rank  2022 Population  2020 Population  2015 Population  2010 Population  \
0      36         41128771         38972230         33753499         28189672   
1     138          2842321          2866849          2882481          2913399   
2      34         44903225         43451666         39543154         35856344   
3     213            44273            46189            51368            54849   
4     203            79824            77700            71746            71519   
..    ...              ...              ...              ...              ...   
229   226            11572            11655            12182            13142   
230   172           575986           556048           491824           413296   
231    46         33696614         32284046         28516545         24743946   
232    63         20017675         18927715         16248230         13792086   
233    74         16320537         15669666         14154937         12839771   

     2000 Population  Growth Rate  World Population Percentage  
0           19542982            1                            0  
1            3182021            0                            0  
2           30774621            1                            0  
3              58230            0                            0  
4              66097            1                            0  
..               ...          ...                          ...  
229            14723            0                            0  
230           270375            1                            0  
231         18628700            1                            0  
232          9891136            1                            0  
233         11834676            1                            0  

[234 rows x 8 columns]

In [16]:
y=df.Continent
y

0      1
1      2
2      3
3      4
4      2
      ..
229    4
230    3
231    1
232    3
233    3
Name: Continent, Length: 234, dtype: int64

# Importing train-test split

In [17]:
from sklearn.model_selection import train_test_split

In [18]:
x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.2)

In [19]:
print(x_train.shape)
print(x_test.shape)
print(y_train.shape)
print(y_test.shape)

(187, 8)
(47, 8)
(187,)
(47,)


# Random Forest

In [20]:
from sklearn.ensemble import RandomForestClassifier

In [21]:
model_rf=RandomForestClassifier(n_estimators=100)

In [22]:
model_rf.fit(x_train,y_train)

RandomForestClassifier()

In [23]:
model_rf.score(x_train,y_train)*100

100.0

In [24]:
model_rf.score(x_test,y_test)*100

31.914893617021278

In [25]:
y_predict=model_rf.predict(x_test)

In [26]:
y_predict

array([3, 1, 2, 2, 4, 5, 5, 2, 3, 1, 1, 4, 3, 2, 1, 3, 6, 2, 1, 5, 1, 2,
       3, 1, 2, 5, 3, 5, 2, 3, 4, 1, 4, 3, 4, 2, 1, 5, 1, 3, 4, 1, 2, 3,
       2, 3, 5], dtype=int64)

In [27]:
df=pd.DataFrame({'Actual':y_test,'Predicted':y_predict})
df

Actual  Predicted
207       1          3
117       2          1
113       3          2
73        1          2
64        6          4
54        5          5
91        2          5
193       3          2
103       2          3
191       4          1
160       4          1
107       4          4
26        3          3
176       5          2
228       1          1
84        3          3
143       1          6
45        5          2
57        3          1
157       4          5
162       6          1
47        5          2
124       3          3
0         1          1
46        2          2
153       4          5
156       1          3
14        5          5
146       4          2
16        1          3
151       1          4
55        3          1
10        5          4
34        3          3
203       1          4
218       2          2
131       5          1
37        5          5
75        3          1
49        2          3
17        5          4
51        2          1
189       2          2
27        6          3
134       2          2
141       3          3
44        4          5

In [28]:
from sklearn.metrics import accuracy_score

In [29]:
test_acc = accuracy_score(y_test,y_predict)*100

In [30]:
test_acc

31.914893617021278

# Classification report

In [31]:
from sklearn.metrics import classification_report

In [32]:
performance=classification_report(y_test,y_predict)

In [33]:
print(performance)

              precision    recall  f1-score   support

           1       0.18      0.22      0.20         9
           2       0.36      0.44      0.40         9
           3       0.45      0.50      0.48        10
           4       0.17      0.14      0.15         7
           5       0.43      0.33      0.38         9
           6       0.00      0.00      0.00         3

    accuracy                           0.32        47
   macro avg       0.27      0.27      0.27        47
weighted avg       0.31      0.32      0.31        47



# Conclusion

Acurracy of training dataset = 100%
Accuracy of testing dataset =  31.91%

As we are getting very high accuracy in training data but very low accuracy in our testing data means over data set ,it is an example of overfiting